### PR TITLE
teku: rebuild with upstream patches

### PR DESCRIPTION
**Pull request by iMichka**
Fixes:
  > Task :infrastructure:kzg:compileJava FAILED
  /private/tmp/teku-20230121-63384-2ktnit/infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java:16: error: cannot find symbol
  import ethereum.ckzg4844.CKzg4844JNI;
                          ^
    symbol:   class CKzg4844JNI
    location: package ethereum.ckzg4844

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by cho-m** [comment]
```
  > Failed to apply plugin 'com.github.jk1.dependency-license-report'.
     > License Report Plugin requires Gradle 7.0. Gradle 6.9.3 detected.
```

May just want to use `./gradlew` instead as this probably needs a specific Gradle version we don't provide.

**Comment by SMillerDev** [comment]
Still doesn't build

**Comment by cho-m** [comment]
May not be Gradle issue then. I did try adding HEAD and didn't see an issue when building locally, so probably something fixed upstream.

File: Formula/teku.rb [hunk 1]
```diff
@@ -34,3 +44,122 @@ def install
...
+diff --git a/gradle/versions.gradle b/gradle/versions.gradle
+index c1916f59d..03e389f71 100644
+--- a/gradle/versions.gradle
++++ b/gradle/versions.gradle
+@@ -41,7 +41,7 @@ dependencyManagement {
+
+     dependency 'io.libp2p:jvm-libp2p-minimal:0.10.0-RELEASE'
+     dependency 'tech.pegasys:jblst:0.3.8'
+-    dependency 'tech.pegasys:jc-kzg-4844:develop'
++    dependency 'tech.pegasys:jc-kzg-4844:0.1.0'
```
Comment (chenrui333) [inline Formula/teku.rb:58]: this is the root cause of the regression build failure.

**Comment by chenrui333** [comment]
Should be good now (passed build/test in my local)

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**